cinema/views.py

```diff
@@ -112,6 +162,24 @@ def get_serializer_class(self):
 
         return MovieSessionSerializer
 
+    def get_permissions(self):
+        if self.action in [
+            "create",
+            "list",
+            "retrieve",
+            "update",
+            "partial_update",
+            "delete",
```

Comment:
The action 'delete' should be 'destroy' in the DRF viewset actions. Update 'delete' to 'destroy' to ensure the correct permission is applied.